src/Schema.php

```diff
@@ -46,6 +49,41 @@ public static function uninstall() {
     Admin::removeAccessCapability();
   }
 
+  /**
+   * Ensures dynamic response headers for all requests from htaccess.
+   */
+  public static function ensureResponseHeadersHtaccess() {
+    $template = file_get_contents(Plugin::getBasePath() . '/conf/.htaccess.headers');
+    if (defined('CORE_STANDARDS_CONTENT_SECURITY_POLICY')) {
+      static::findAndReplaceHeader($template, 'Content-Security-Policy', CORE_STANDARDS_CONTENT_SECURITY_POLICY);
+    }
+    if (defined('CORE_STANDARDS_REFERRER_POLICY')) {
+      static::findAndReplaceHeader($template, 'Referrer-Policy', CORE_STANDARDS_REFERRER_POLICY);
+    }
+    if (defined('CORE_STANDARDS_STRICT_TRANSPORT_SECURITY')) {
+      static::findAndReplaceHeader($template, 'Strict-Transport-Security', CORE_STANDARDS_STRICT_TRANSPORT_SECURITY);
+    }
+    if (defined('CORE_STANDARDS_X_CONTENT_TYPE_OPTIONS')) {
+      static::findAndReplaceHeader($template, 'X-Content-Type-Options', CORE_STANDARDS_X_CONTENT_TYPE_OPTIONS);
+    }
+    if (defined('CORE_STANDARDS_X_XSS_PROTECTION')) {
+      static::findAndReplaceHeader($template, 'X-XSS-Protection', CORE_STANDARDS_X_XSS_PROTECTION);
```

Comment:
1. Considering the actual usage in `wp-config.php`, it would probably be cleaner and easier to specify all of the value overrides in a single constant instead of separate constants, whose names you don't know and first have to look up?
    ```php
    const CORE_STANDARDS_X_CONTENT_TYPE_OPTIONS = '...';
    const CORE_STANDARDS_X_XSS_PROTECTION = '...';
    ```
    vs.
    ```php
    const CORE_STANDARDS_SECURITY_HEADERS = [
      'X-Content-Type-Options' => '...',
      'X-XSS-Protection' => '...',
    ];
    ```

2. What happens if a site updates to the new code and does not have any overrides configured?

3. Looking at [examples of actual production values](https://github.com/netzstrategen/nutr-nmed/blob/master/.htaccess#L194), the values in wp-config.php will actually get very (very) long.  Values in .htaccess can't use newlines without awkward workarounds.  We could make our lives easier by allowing newlines in values specified in wp-config.php and automatically transforming them into a single line (simply replacing any newlines with spaces).
    ```php
    $header_value = preg_replace('\r?\n', ' ', $header_value);
    ```

Reply:
This is how the config file may look when implementing the extension of this function: 
```
const CORE_STANDARDS_HTTP_HEADERS = [
  'Strict-Transport-Security' => '"max-age=63072000; includeSubDomains; preload" env=HTTPS',
  'X-Frame-Options' => '"ALLOW-FROM https://tools.nutricia.de https://doo.net https://www.youtube.com https://vimeo.com https://games.focusgames.co.uk"',
];
```

So I think is not an issue to keep it like this for special cases only